examples/pipelines/datacomp/components/cluster_image_embeddings/src/main.py

```diff
@@ -0,0 +1,44 @@
+"""This component applies k-means clustering on image embeddings.
+"""
+import logging
+
+import pandas as pd
+import numpy as np
+from sklearn.cluster import MiniBatchKMeans
+
+from fondant.component import PandasTransformComponent
+
+logger = logging.getLogger(__name__)
+
+
+class ClusterImageEmbeddingsComponent(PandasTransformComponent):
+    """Component that clusters images based on embeddings."""
+
+    def setup(self, *, num_clusters: int) -> None:
+        self.kmeans = MiniBatchKMeans(
+            n_clusters=num_clusters, random_state=0, batch_size=6, n_init="auto"
+        )
+
+    def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
+        """
+        Args:
+            dataframe: Pandas dataframe
+
+        Returns:
+            Pandas dataframe
+        """
+        embeddings = dataframe["image"]["embedding"]
```

Comment:
Regarding your question in the daily: Would there be any option to load the column names in generic way? If the component always consumes a dataframe with one subset/column the naming shouldn't matter?

Reply:
Good question. I see 2 mechanisms we could leverage:
- Generic components. These read their data based on the component specification, which is the only thing the user needs to provide. This is already used for the `load_from_hf_hub` and `write_to_hf_hub` components.
- Column mapping between components. We don't have this yet, but are planning to implement this in #180 

Not sure how easy it would be to implement this component in a generic way, but it can be a good test.